webgpu.h

```diff
@@ -0,0 +1,1025 @@
+    WGPU_SHADER_STAGE_BIT_FRAGMENT = 0x00000002,
+    WGPU_SHADER_STAGE_BIT_COMPUTE = 0x00000004,
+    WGPU_SHADER_STAGE_BIT_FORCE32 = 0x7FFFFFFF
+} wgpuShaderStageBit;
+
+typedef enum {
+    WGPU_TEXTURE_USAGE_BIT_NONE = 0x00000000,
+    WGPU_TEXTURE_USAGE_BIT_COPY_SRC = 0x00000001,
+    WGPU_TEXTURE_USAGE_BIT_COPY_DST = 0x00000002,
+    WGPU_TEXTURE_USAGE_BIT_SAMPLED = 0x00000004,
+    WGPU_TEXTURE_USAGE_BIT_STORAGE = 0x00000008,
+    WGPU_TEXTURE_USAGE_BIT_OUTPUT_ATTACHMENT = 0x00000010,
+    WGPU_TEXTURE_USAGE_BIT_FORCE32 = 0x7FFFFFFF
+} wgpuTextureUsageBit;
+
+
+typedef struct wgpuBindGroupBinding {
+    uint32_t binding;
+    wgpuBuffer buffer;
+    uint64_t offset;
+    uint64_t size;
+    wgpuSampler sampler;
+    wgpuTextureView textureView;
+} wgpuBindGroupBinding;
+
+typedef struct wgpuBindGroupLayoutBinding {
+    uint32_t binding;
+    wgpuShaderStageBit visibility;
+    wgpuBindingType type;
+    bool dynamic;
+    bool multisampled;
+    wgpuTextureViewDimension textureDimension;
+} wgpuBindGroupLayoutBinding;
+
+typedef struct wgpuBlendDescriptor {
+    wgpuBlendOperation operation;
+    wgpuBlendFactor srcFactor;
+    wgpuBlendFactor dstFactor;
+} wgpuBlendDescriptor;
+
+typedef struct wgpuBufferCopyView {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuBuffer buffer;
+    uint64_t offset;
+    uint32_t rowPitch;
+    uint32_t imageHeight;
+} wgpuBufferCopyView;
+
+typedef struct wgpuBufferDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuBufferUsageBit usage;
+    uint64_t size;
+} wgpuBufferDescriptor;
+
+typedef struct wgpuColor {
+    float r;
+    float g;
+    float b;
+    float a;
+} wgpuColor;
+
+typedef struct wgpuCommandBufferDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+} wgpuCommandBufferDescriptor;
+
+typedef struct wgpuCommandEncoderDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+} wgpuCommandEncoderDescriptor;
+
+typedef struct wgpuComputePassDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+} wgpuComputePassDescriptor;
+
+typedef struct wgpuCreateBufferMappedResult {
+    wgpuBuffer buffer;
+    uint64_t dataLength;
+    void * data;
+} wgpuCreateBufferMappedResult;
+
+typedef struct wgpuExtent3D {
+    uint32_t width;
+    uint32_t height;
+    uint32_t depth;
+} wgpuExtent3D;
+
+typedef struct wgpuFenceDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    uint64_t initialValue;
+} wgpuFenceDescriptor;
+
+typedef struct wgpuOrigin3D {
+    uint32_t x;
+    uint32_t y;
+    uint32_t z;
+} wgpuOrigin3D;
+
+typedef struct wgpuPipelineLayoutDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    uint32_t bindGroupLayoutCount;
+    wgpuBindGroupLayout const * bindGroupLayouts;
+} wgpuPipelineLayoutDescriptor;
+
+typedef struct wgpuPipelineStageDescriptor {
+    const void* nextInChain;
```

Comment:
can we be consistent here, i.e. `void const * nextInChain` ?

Reply:
Done